### PR TITLE
Linters and refactoring

### PR DESCRIPTION
**Pull request by knopki**
- add `dependabot` configuration
- configure linters and add them as optional dependencies
- create github action workflow
- downgrade code to python3.8
- make linter happy
- refactor all DTOs and model to match https://github.com/basf/metis-gui/blob/master/src/types/dto.d.ts (this was a mistake - too many errors in typescript types)
- make models and dto more uniform

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
```
Comment (blokhin) [inline .github/dependabot.yml:1]: Should not this file be in the `workflows` folder?

File: metis_client/models/calculation.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "Calculation model"
+    id: int
+    name: str
+    user_id: int
+    status: int
+    progress: int
+    result: Sequence[MetisDataSourceModel]
+
+    @classmethod
+    def from_dto(cls, dto: MetisCalculationDTO) -> "MetisCalculationModel":
```
Comment (blokhin) [inline metis_client/models/calculation.py:31]: The similar`from_dto` method is used across many other places; would it make sense to add an inheritable mixture maybe instead?

**Review by blokhin — APPROVED** [review]

File: metis_client/client.py [hunk 3]
```diff
@@ -282,15 +299,15 @@ async def sse(
...
+                        "%s - authentication requered - reconnecting after reauth",
+                        self._base_url,
                     )
                     await self._do_auth(force=True)
                     continue
                 raise
 
             except ConnectionAbortedError as err:
-                self.logger.error(f"{self._base_url} connection error {err} - abort")
+                self.logger.error("%s connection error %s - abort", self._base_url, err)
```
[the remark this answers is not in the document]
Comment (knopki) [reply metis_client/client.py:310]: We can, this is just a `pylint` recommendation.

File: metis_client/models/collection.py [hunk 4]
```diff
@@ -1,42 +1,124 @@
...
+
 from dataclasses import dataclass
-from datetime import datetime
-from typing import TypedDict, cast
+from typing import Literal, Optional, Sequence, Union
+
+from .timestamp import MetisTimestampsDTO, MetisTimestampsModel
+from .user import MetisUserOnlyNameEmailDTO, MetisUserOnlyNameEmailModel
+
+MetisCollectionFlavor = Union[
```
[the remark this answers is not in the document]
Comment (knopki) [reply metis_client/models/collection.py:9]: Really? I was wondering what the weird thing was :)
Ok, let it be just a string.

File: .github/dependabot.yml (hunk 1 above)
Comment (knopki) [reply .github/dependabot.yml:1]: No, it is [dependabot](https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/) configuration, not a workflow.

File: metis_client/models/calculation.py (hunk 2 above)
Comment (knopki) [reply metis_client/models/calculation.py:31]: But all `from_dto`'s are different.

File: metis_client/models/collection.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (knopki) [reply metis_client/models/collection.py]: Fixed in https://github.com/tilde-lab/metis-client/pull/3/commits/f0c51fa810bbfd5f8304b909fd63a49c6e1be64b

**Comment by knopki** [comment]
@blokhin Can we enable Github Actions in this repo?

File: metis_client/client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (blokhin) [reply metis_client/client.py:310]: I think, it recommends that for a reason, I remember, I read somewhere about an immediate and postponed execution in these two string substitution types 🤔

**Pull request merged**